enterprise/internal/authz/github/github.go

```
@@ -539,6 +540,23 @@ func (p *Provider) getRepoAffiliatedGroups(ctx context.Context, owner, name stri
 		groups = append(groups, repoAffiliatedGroup{cachedGroup: group, adminsOnly: adminsOnly})
 	}
 
+	// The visibility field on a repo is only returned if this feature flag is set. As a result
+	// there's no point in making an extra API call if this feature flag is not set explicitly.
+	if conf.ExperimentalFeatures().EnableGithubInternalRepoVisibility {
+		var r *github.Repository
+		r, err = p.client.GetRepository(ctx, owner, name)
+		if err != nil {
+			// Maybe the repo doesn't belong to this org? Or Another error occurred in trying to get the
+			// repo. Either way, we are not going to syncGroup for this repo.
+			return
+		}
```

Should we debug log this error?

---

I traced the code path of the error being returned all the way to the call site of `FetchRepoPerms` and ultimately `syncRepoPerms` and `syncPerms` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/cmd/repo-updater/internal/authz/perms_syncer.go?L905:12&subtree=true), where we end up logging the error. So I think we don't need to debug log this explicitly. If there's an error because of this it would show up in the logs? :thinking:

---

I almost thought we ignored the error because of the naked return :)